tests/v2/validation/charts/cisbenchmark_test.go

```diff
@@ -0,0 +1,298 @@
+	initialCisbenchmarkChart, err := charts.GetChartStatus(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	if initialCisbenchmarkChart.IsAlreadyInstalled && initialCisbenchmarkChart.ChartDetails.Spec.Chart.Metadata.Version == versionLatest {
+		c.T().Skip("Skipping the upgrade case, cis-benchmark chart is already installed with the latest version")
+	}
+
+	if !initialCisbenchmarkChart.IsAlreadyInstalled {
+		c.T().Log("Installing cis-benchmark chart with the version before the latest version")
+		err = charts.InstallRancherCisBenchmarkChart(client, c.chartInstallOptions)
+		require.NoError(c.T(), err)
+
+		c.T().Log("Waiting cis-benchmark chart deployments to have expected number of available replicas")
+		err = charts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+		require.NoError(c.T(), err)
+
+		c.T().Log("Waiting cisbenchmark chart DaemonSets to have expected number of available nodes")
+		err = charts.WatchAndWaitDaemonSets(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+		require.NoError(c.T(), err)
+	}
+
+	cisbenchmarkChartPreUpgrade, err := charts.GetChartStatus(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	// Validate current version of rancher-cis-benchmark is one of the versions before latest
+	chartVersionPreUpgrade := cisbenchmarkChartPreUpgrade.ChartDetails.Spec.Chart.Metadata.Version
+	require.Contains(c.T(), versionsList[1:], chartVersionPreUpgrade)
+
+	c.chartInstallOptions.Version, err = client.Catalog.GetLatestChartVersion(charts.RancherCisBenchmarkName, catalog.RancherChartRepo)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Upgrading cis-benchmark chart to the latest version")
+	err = charts.UpgradeRancherCisBenchmarkChart(client, c.chartInstallOptions)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Waiting for cis-benchmark chart deployments to have expected number of available replicas after upgrade")
+	err = charts.WatchAndWaitDeployments(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+	require.NoError(c.T(), err)
+
+	c.T().Log("Waiting cis-benchmark chart DaemonSets to have expected number of available nodes after upgrade")
+	err = charts.WatchAndWaitDaemonSets(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, metav1.ListOptions{})
+	require.NoError(c.T(), err)
+
+	cisbenchmarkChartPostUpgrade, err := charts.GetChartStatus(client, c.project.ClusterID, charts.RancherCisBenchmarkNamespace, charts.RancherCisBenchmarkName)
+	require.NoError(c.T(), err)
+
+	c.T().Log("Comparing installed and desired cis-benchmark versions")
+	chartVersionPostUpgrade := cisbenchmarkChartPostUpgrade.ChartDetails.Spec.Chart.Metadata.Version
+	require.Equal(c.T(), c.chartInstallOptions.Version, chartVersionPostUpgrade)
```

Comment:
I think a scan should be taken after the upgrade to ensure the chart is working as expected after the upgrade. Perhaps even viewing an existing scan to ensure nothing was lost / affected from previous usage after an upgrade has occurred.

Reply:
Yes, I updated the code to run the scan after upgrade.